packages/playwright-core/src/server/fetch.ts

```diff
@@ -307,8 +309,27 @@ export abstract class APIRequestContext extends SdkObject {
       // If we have a proxy agent already, do not override it.
       const agent = options.agent || (url.protocol === 'https:' ? httpsHappyEyeballsAgent : httpHappyEyeballsAgent);
       const requestOptions = { ...options, agent };
+
+      const startAt = monotonicTime();
+      let dnsLookupAt: number | undefined;
+      let tcpConnectionAt: number | undefined;
+      let tlsHandshakeAt: number | undefined;
+      let requestFinishAt: number | undefined;
+      let firstByteAt: number | undefined;
+      let endAt: number | undefined;
+
       const request = requestConstructor(url, requestOptions as any, async response => {
         const notifyRequestFinished = (body?: Buffer) => {
+          const timings: har.Timings = {
+            send: requestFinishAt! - startAt,
+            wait: firstByteAt! - requestFinishAt!,
```

Comment:
What if body was empty, and we did not set `firstByteAt`?

Reply:
Good catch. I've replaced `firstByteAt` with a measure of when the response headers are sent in 5d8ab8a33cde506ca1076fe349fd530870c17b69. I think that's more in-line with the HAR definition anyways.